files/rsyslog/67-mmexternal.conf

```diff
@@ -0,0 +1,15 @@
+if (strlen(`echo $USE_MMEXTERNAL`) > 0) and (`echo $USE_MMEXTERNAL` == "true") then {
+    module(load="mmexternal")
+    action(type="mmexternal" binary="/usr/local/bin/undefined_field" interface.input="fulljson")
```

Comment:
so `/usr/local/bin/undefined_field` also does the skip empty field handling?

Reply:
Yes, it does.  And undefined_field skip empty supports CDM_KEEP_EMPTY_FIELDS, while the else part mmnormalization with parse_json_skip_empty.rulebase does not.

The reason why I added USE_MMEXTERNAL and SKIP_EMPTY is for the performance measurement.  Just in case mmexternal shows degradation, we could provide Plan B to the people who don't care the undefined and CDM_KEEP_EMPTY_FIELDS.  (Or we could completely turn it off.)